graph/telemetry/istio/appender/ambient.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 const AmbientAppenderName = "ambient"
-const WaypointSuffix = "-istio-waypoint"
+const WaypointSuffix = "waypoint"
 
 type AmbientAppender struct {
```

Comment:
Updated comment. 

I think the appender makes sense because it isolates the Ambient logic in just one place. It also is adding some logic regarding identifying the waypoint nodes and also removing the edged. I've also added another change to prevent this error:

Reply:
Or, if the graph generator uses the node (Or it has access to the labels) It could mark waypoint as the metadata (And later, it would be easier to process)

Reply:
The graph-gen code drives off of telemetry metrics and does not perform any k8s api queries.  So, as I suspected above, and you confirmed, I think the appender approach is required to handle the `showWaypoints` option.